### ISSUE TITLE
Re-thinking some reports

### ISSUE THREAD

**Issue by oesteban** [poldracklab/fmriprep#890]
After reviewing the 60 reports I had, I got to the conclusion that we might want to decouple reports and interfaces.

The problem is that interfaces take very specific inputs that are probably not the best option for visual inspection. On the other hand, the more you decouple, the harder it is to use the reportlet to be sure an error happened in a certain interface.

However, the reports are meant for the researchers to check on the quality of preprocessing. For instance, having the enhanced BOLD image flickering with the T1 image does not help much because drop out areas are exaggerated. For this plot I would render the average after corrections (STC, HMC, SDC but no enhancements). Same for the SyN plot, which could also use the brain mask contour.

WDYT?

**Comment by effigies** [comment]
I think this is reasonable. The failing interfaces are dropping crash files in the output directory, so a missing report isn't our biggest clue that something's going wrong, anymore. Setting up independent visualization nodes will also make it easier to account for the resource usage of the visualizations.

**Comment by oesteban** [comment]
Yep, that was another reason I missed to reflect.

**Comment by chrisfilo** [comment]
On Thu, Dec 14, 2017 at 11:02 AM, Oscar Esteban <notifications@github.com>
wrote:

> After reviewing the 60 reports I had, I got to the conclusion that we
> might want to decouple reports and interfaces.
>
> The problem is that interfaces take very specific inputs that are probably
> not the best option for visual inspection. On the other hand, the more you
> decouple, the harder it is to use the reportlet to be sure an error
> happened in a certain interface.
>
I think it depends on the use case. There is a place for both and AFAIK we
already have dedicated report generation nodes in FMRIPREP where they were
needed.

> However, the reports are meant for the researchers to check on the quality
> of preprocessing. For instance, having the enhanced BOLD image flickering
> with the T1 image does not help much because drop out areas are
> exaggerated. For this plot I would render the average after corrections
> (STC, HMC, SDC but no enhancements).
>
Maybe I'm missing something, but the signal dropouts should be outside of
the brain mask (and we are plotting masked images) and the contrast
enhancement increases tissue contrast so it should make evaluating
coregistration easier. It's possible that a combination of bad
skullstripping and poor enhancement lead to weird image that threw off
coregistration, but we need to see it to be able to assess the problem.

> Same for the SyN plot, which could also use the brain mask contour.
>
The SyN SDC or T1-to-MNI Syn?

> WDYT?
>
In short I would use dedicated visualization nodes where there is a
specific need for them.

**Comment by emdupre** [comment]
I'd also like to chime in multi-echo could benefit from decoupling the reports a bit more, in the sense that I've been envisioning a drop-down button for the functional run summary, where users would select the echo they were interested in seeing-- something that I don't think is possible at the moment. 

Right now, the reports are generated and compiled for each echo individually, which (especially since they can appear out of order) makes it hard to visually understand how well everything performed. An example report is shared here: https://www.dropbox.com/sh/ip6xpd6oc5awnbm/AADp92KIKev9h6Yb8fIZsYu7a?dl=0

**Comment by oesteban** [comment]
Let's follow up discussion on the multi-echo data in #891.
